### PR TITLE
Run HPU tests only with yml (#12469)

### LINKED ISSUE #12469: HPU CI error File exists: /azp/agent/_work/1/s/test_temp
## 🐛 Bug

After tests have passed, the CI errors when saving a temp file. 

### To Reproduce

[See CI output.](https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=63944&view=logs&j=4d03c832-dbbd-5507-ad7a-9e2939b77985&t=1836f526-443b-5c5d-ee99-43d073ed9dba&l=108)

```
Printing error_log.txt...
synapse_logger synapse_logger INFO. pid=INFO. pid=1939272 at 1939273/home/jenkins/workspace/cdsoftwarebuilder/create-pytorch---bpt-d/repos/pytorch-integration/pytorch_helpers/synapse_logger/synapse_logger.cpp at :/home/jenkins/workspace/cdsoftwarebuilder/create-pytorch---bpt-d/repos/pytorch-integration/pytorch_helpers/synapse_logger/synapse_logger.cpp340: 340Done command:  restartDone command: 
restart
ImportError while loading conftest '/azp/agent/_work/1/s/tests/conftest.py'.
tests/__init__.py:34: in <module>
    os.mkdir(_TEMP_PATH)
E   FileExistsError: [Errno 17] File exists: '/azp/agent/_work/1/s/test_temp

```

### Environment

HPU CI machine

cc @carmocca @akihironitta @borda @tchaton @rohitgr7

### PR DESCRIPTION
**Pull request by jerome-habana**
Execute supported tests serially

Signed-off-by: Akarsha <arao@habana.ai>

This will also resolve #12469 

## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .azure-pipelines/hpu-tests.yml [hunk 1]
```diff
@@ -37,8 +37,16 @@ jobs:
       displayName: 'Install dependencies'
 
     - bash: |
-         python ".azure-pipelines/run_hpu_tests.py"
-      displayName: 'HPU Tests in parallel'
+        python -m pytest -sv tests/accelerators/test_hpu.py --forked --junitxml=hpu1_test-results.xml
+      displayName: 'Single card HPU test'
+
+    - bash: |
+        python -m pytest -sv tests/accelerators/test_hpu.py --forked --hpus 8 --junitxml=hpu8_test-results.xml
```
Comment (kaushikb11) [inline .azure-pipelines/hpu-tests.yml:44]: This would also a lot of tests twice which do not require the `hpus` param

**Review by Borda — APPROVED** [review]
seems finally going the direction I was asking since beginning :rabbit:

**Review by kaushikb11 — APPROVED** [review]

File: .azure-pipelines/hpu-tests.yml (hunk 1 above)
Comment (jerome-habana) [reply .azure-pipelines/hpu-tests.yml:44]: I'll parameterize it along with next change

**Review by akihironitta — APPROVED** [review]
LGTM!

**Pull request merged**